kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveStoreRegistration.java

```diff
@@ -1,42 +1,142 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package dev.responsive.kafka.store;
 
 import com.datastax.oss.driver.shaded.guava.common.annotations.VisibleForTesting;
 import java.util.Objects;
+import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
+import java.util.function.LongSupplier;
 import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
 
 public final class ResponsiveStoreRegistration {
-  private final TopicPartition changelogTopicPartition;
-  private final long committedOffset;
+  public static final long NO_COMMITTED_OFFSET = -1L; // buffer is initialized but no prior offset
+  public static final long UNINITIALIZED_OFFSET = -2L; // buffer is not yet initialized
+
+  private final Logger log;
   private final String name;
+  private final TopicPartition changelogTopicPartition;
   private final Consumer<Long> onCommit;
 
+  private final LongSupplier storedOffset; // RPC to fetch offset committed in remote table
```

Comment:
nit: it might be better to have the stored offset supplier (or just the stored offset) returned by the initialize callback iff the store is active. That way you're guaranteed at compile time that we never query the offset on a standby store.